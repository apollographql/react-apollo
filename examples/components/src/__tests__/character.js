import React from 'react';

import { render } from 'react-testing-library';
import Character from '../character';

import {
  empty,
  hero_no_friends,
  empty_array_friends,
  friend_without_appearsIn,
} from '../__mocks__/data';

describe('Character', () => {
  it('returns markup for null response', () => {
    const { container } = render(<Character hero={empty} />);
    expect(container).toMatchSnapshot();
  });
  it('returns markup for a hero with no friends', () => {
    const { container } = render(<Character hero={hero_no_friends} />);
    expect(container).toMatchSnapshot();
  });
  it('returns markup for empty array of friends', () => {
    const { container } = render(<Character hero={empty_array_friends} />);
    expect(container).toMatchSnapshot();
  });
  it('returns markup for a friend without an appearsIn', () => {
    const { container } = render(<Character hero={friend_without_appearsIn} />);
    expect(container).toMatchSnapshot();
  });
});
