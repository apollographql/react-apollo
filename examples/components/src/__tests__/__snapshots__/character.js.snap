// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character returns markup for a friend without an appearsIn 1`] = `
<div>
  <div>
    <div>
      <h3>
        r2d2
      </h3>
      <h6>
        luke
        : 
        newhope
      </h6>
      <h6>
        james
        : 
      </h6>
    </div>
  </div>
</div>
`;

exports[`Character returns markup for a hero with no friends 1`] = `
<div>
  <div>
    <div>
      <h3>
        r2d2
      </h3>
    </div>
  </div>
</div>
`;

exports[`Character returns markup for empty array of friends 1`] = `
<div>
  <div>
    <div>
      <h3>
        r2d2
      </h3>
    </div>
  </div>
</div>
`;

exports[`Character returns markup for null response 1`] = `
<div>
  <div />
</div>
`;
